presto-accumulo/src/main/java/com/facebook/presto/accumulo/io/AccumuloRecordCursor.java

```diff
@@ -0,0 +1,537 @@
+                    return false;
+                }
+            }
+
+            // If we do have key/value pairs to process from Accumulo, we reset and deserialize the
+            // previously scanned key/value pair as we have started a new row.
+            // This code block does not execute the first time this method is called
+            if (prevRowID.getLength() != 0) {
+                serializer.reset();
+                serializer.deserialize(prevKV);
+            }
+
+            // While there are key/value pairs to read and we have not advanced to a new row
+            boolean advancedToNewRow = false;
+            while (iterator.hasNext() && !advancedToNewRow) {
+                // Scan the key value pair and get the row ID
+                Entry<Key, Value> kv = iterator.next();
+
+                bytesRead += kv.getKey().getSize() + kv.getValue().getSize();
+                kv.getKey().getRow(rowID);
+
+                // If the row IDs are equivalent or there is no previous row
+                // ID (length == 0), deserialize the KV pair
+                if (rowID.equals(prevRowID) || prevRowID.getLength() == 0) {
+                    serializer.deserialize(kv);
+                }
+                else {
+                    // If they are different, then we have advanced to a new row and we need to
+                    // break out of the loop
+                    advancedToNewRow = true;
+                }
+
+                // Set our 'previous' member variables to track the previously scanned entry
+                prevKV = kv;
+                prevRowID.set(rowID);
+            }
+
+            // If we didn't break out of the loop, we have reached the last entry in our
+            // BatchScanner, so we set this to null as the entire row has been processed
+            // This tracks the edge case where it is one entry per row ID
+            if (!advancedToNewRow) {
+                prevKV = null;
+            }
+
+            // Return true so Presto will process the row we've just ran
+            return true;
+        }
+        catch (IOException e) {
+            throw new PrestoException(IO_ERROR, "Caught IO error from serializer on read", e);
+        }
+    }
+
+    @Override
+    public boolean isNull(int field)
+    {
+        checkArgument(field < cHandles.size(), "Invalid field index");
+        return serializer.isNull(fieldToColumnName[field]);
+    }
+
+    @Override
+    public boolean getBoolean(int field)
+    {
+        checkFieldType(field, BOOLEAN);
+        return serializer.getBoolean(fieldToColumnName[field]);
+    }
+
+    @Override
+    public double getDouble(int field)
+    {
+        checkFieldType(field, DOUBLE);
+        return serializer.getDouble(fieldToColumnName[field]);
+    }
+
+    @Override
+    public long getLong(int field)
+    {
+        checkFieldType(field, BIGINT, DATE, INTEGER, TIME, TIMESTAMP);
+        switch (getType(field).getDisplayName()) {
```

Comment:
I've been changing the places where I switched on display names to an `if/else` blocks using `type.equals(BIGINT)`, etc., to be more consistent on how I match on types.